pkg/api/api.go

```diff
@@ -1,63 +1,77 @@
 package api
 
 import (
-	"errors"
-	"io"
 	"net/http"
 	"os"
 
 	log "github.com/sirupsen/logrus"
 )
 
-var (
-	lock chan bool
-)
+const tokenMissingMsg = "api token is empty or has not been set. exiting"
 
-func init() {
-	lock = make(chan bool, 1)
-	lock <- true
+// API is the http server responsible for serving the HTTP API endpoints
+type API struct {
+	Token       string
+	hasHandlers bool
 }
 
-// SetupHTTPUpdates configures the endpoint needed for triggering updates via http
-func SetupHTTPUpdates(apiToken string, updateFunction func()) error {
-	if apiToken == "" {
-		return errors.New("api token is empty or has not been set. not starting api")
+// New is a factory function creating a new API instance
+func New(token string) *API {
+	return &API{
+		Token:       token,
+		hasHandlers: false,
 	}
+}
 
-	log.Println("Watchtower HTTP API started.")
-
-	http.HandleFunc("/v1/update", func(w http.ResponseWriter, r *http.Request) {
-		log.Info("Updates triggered by HTTP API request.")
-
-		_, err := io.Copy(os.Stdout, r.Body)
-		if err != nil {
-			log.Println(err)
+// RequireToken is wrapper around http.HandleFunc that checks token validity
+func (api *API) RequireToken(fn http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("Token") != api.Token {
+			log.Error("Invalid token.")
 			return
 		}
+		log.Println("Valid token found.")
+		fn(w, r)
+	}
+}
 
-		if r.Header.Get("Token") != apiToken {
-			log.Println("Invalid token. Not updating.")
-			return
-		}
+// RegisterFunc is a wrapper around http.HandleFunc that also flips the bool used to determine whether to launch the API
```

Comment:
```suggestion
// RegisterFunc is a wrapper around http.HandleFunc that also sets the flag used to determine whether to launch the API
```
That would obviously be an insane way to do an API, but it sounds like it would just toggle it back the second time